src/main/java/core/basesyntax/service/impl/CsvParserService.java

```diff
@@ -0,0 +1,62 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ParserService;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CsvParserService implements ParserService {
+    @Override
+    public List<FruitTransaction> parseCsv(List<String> lines) {
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+
+        for (String line : lines) {
+            FruitTransaction transaction = parseCsvLine(line);
+            fruitTransactions.add(transaction);
+        }
+
+        return fruitTransactions;
+    }
+
+    private FruitTransaction parseCsvLine(String line) {
+        String[] parsedArray = line.split(",");
+
+        if (parsedArray.length == 3 && !parsedArray[0].equalsIgnoreCase("type")) {
+            String operationCode = parsedArray[0].trim();
+            String fruit = parsedArray[1].trim();
+
+            try {
+                int quantity = Integer.parseInt(parsedArray[2].trim());
+
+                FruitTransaction.Operation operation = mapOperation(operationCode);

```

Comment:
add method to the enum
```suggestion
                FruitTransaction.Operation operation = FruitTransaction.Operation.getByCode(operationCode);
```